parsel.ts

```diff
@@ -406,33 +429,42 @@ export function specificity(selector: string | AST): number[] {
 		return specificities[numbers.indexOf(Math.max(...numbers))];
 	}
 
-	let ret = [0, 0, 0];
-	walk(ast, (node) => {
-		if (node.type === 'id') {
-			ret[0]++;
-		} else if (node.type === 'class' || node.type === 'attribute') {
-			ret[1]++;
-		} else if (
-			(node.type === 'type' && node.content !== '*') ||
-			node.type === 'pseudo-element'
-		) {
-			ret[2]++;
-		} else if (node.type === 'pseudo-class' && node.name !== 'where') {
-			if (RECURSIVE_PSEUDO_CLASSES.has(node.name) && node.subtree) {
-				const sub = specificity(node.subtree);
+	const ret = [0, 0, 0];
+	for (const [token] of flatten(ast)) {
+		switch (token.type) {
+			case 'id':
+				ret[0]++;
+				break;
+			case 'class':
+			case 'attribute':
+				ret[1]++;
+				break;
+			case 'pseudo-element':
+			case 'type':
+				ret[2]++;
```

Comment:
This is not an equivalent condition: note that the original code also checks `node.content` if `node.type` is `"type"`.

Reply:
Because of the `universal` types are distinct type now, we can get rid of that condition :)

Reply:
> I'm also not sure why to change node to token?

By habit. For this case, I named it token so it's clear we are handling tokens, not nodes of an AST.